src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java

```diff
@@ -1305,6 +1307,26 @@ public static String firstPage(String pages) {
     }
 
     /**
+     * Return the non-digit prefix of pages
+     *
+     * @param pages
+     *            a pages string such as L42--111 or L7,41,73--97 or L43+
+     *
+     * @return the non-digit prefix of pages (like "L" of L7)
+     *         or "" if no non-digit prefix is found in the string
+     *
+     * @throws NullPointerException
+     *             if pages is null.
+     */
+    public static String pagePrefix(String pages) {
+        if (pages.matches("^\\D+.*$")) {
```

Comment:
It is better if you extract the regex to a constant pattern, by using Pattern.compile  , have a look at the titleChecker (logic.integrity package ) class, to see how it's done.
This construct compiles the regex and makes it faster when it is called multiple times